jena-base/src/main/java/org/apache/jena/atlas/lib/cache/CacheSimple.java

```diff
@@ -37,120 +37,81 @@
 
 public class CacheSimple<K,V> implements Cache<K,V>
 {
-    private final V[] values ; 
-    private final K[] keys ;
-    private final int size ;
-    private int currentSize = 0 ;
-    private BiConsumer<K,V> dropHandler = null ;
+	private Map<K,V> internalCache;
+	
```

Comment:
If keys K1 and K2 have the same short hash, then one adding K2 overwrites k1 in CacheSimple but in a java Map, you end up with two entries K1 and K2, with internal reorganisation as needed and the Map is bigger. CacheSimple is a sort of fixed size map.